### PR TITLE
fixing unboxing of categoricals in DFs

### PR DESCRIPTION
**Pull request by fschlimb**
@anmyachev

### CODE REVIEW

File: hpat/tests/test_dataframe.py [hunk 1]
```diff
@@ -122,6 +122,18 @@ def test_impl(df):
         df = pd.DataFrame({'A': ['aa', 'bb', 'cc']})
         pd.testing.assert_frame_equal(hpat_func(df), test_impl(df))
 
+    def test_box_categorical(self):
+        def test_impl(df):
+            df['A'] = df['A'] + 1
+            return df
+
+        hpat_func = hpat.jit(test_impl)
+        from pandas.api.types import CategoricalDtype
```
Comment (shssf) [inline hpat/tests/test_dataframe.py]: Do we really need this import inside the function? If not, I would prefer to see it at the top of file.

Comment (fschlimb) [reply hpat/tests/test_dataframe.py]: We do not need this symbol anywhere else, so why expose it at global scope? Why import unless needed in the first place? I think this is cleaner (a potential minor performance impact if called multiple times can be ignore here).

Comment (shssf) [reply hpat/tests/test_dataframe.py]: @fschlimb Not importing files at top of the file is a PEP8 violation. We need visible reason ("why do it in regular way" is not acceptable) to do this.

Comment (fschlimb) [reply hpat/tests/test_dataframe.py]: Then no import...

**Pull request merged**